src/main/AuthenticationResponse.ts

```diff
@@ -141,24 +117,12 @@ export default class AuthenticationResponse {
     } else if (!verifyOpts.verification.checkLinkedDomain) {
       await validateLinkedDomainWithDid(issuerDid, verifyOpts.verifyCallback, CheckLinkedDomain.IF_PRESENT);
     }
-    const verPayload = verifiedJWT.payload as AuthenticationResponsePayload;
+    const verPayload = verifiedJWT.payload as IdToken;
```

Comment:
This doesn't seem right. The payload contained the id_token value in line 84, so how can you cast the whole payload as IdToken Payload?

Reply:
It comes from line 110, at this point payload is a Partial<JWTPayload>. Since IdToken extends JWTPayload I can safely cast it to IdToken, besides the cast is necessary because assertValidResponseJWT(...) is expecting and IdToken object;